components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -107,6 +107,17 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         //check if the request is a handshake
         if (msg instanceof FullHttpRequest) {
             FullHttpRequest req = (FullHttpRequest) msg;
+
+            // This block is for the health check of the ports 8099 and 9099
+            if (!req.headers().contains(HttpHeaders.UPGRADE)) {
+                FullHttpResponse httpResponse =
+                        new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+                httpResponse.headers().set("content-type", "text/plain; charset=UTF-8");
+                httpResponse.headers().set("content-length", httpResponse.content().readableBytes());
+                ctx.writeAndFlush(httpResponse);
+                return;
```

Comment:
since we are returning after adding to InboundMesssageContextMap, need to remove the requestContext from map before returning.

```suggestion
             InboundMessageContextDataHolder.getInstance().removeInboundMessageContextForConnection(channelId);
```

IMO this health check header check and return should be done before #L90

Reply:
If we are adding it before L90, we have to duplicate the condition `if (msg instanceof FullHttpRequest)` as well. will that be ok?

Reply:
IMO thats okay instead of adding the inboundcontext unnecessarily to the map and setting analytics properties to the channel.  If check should be a combination of httprequest header check and upgrade header check ```if (msg instanceof FullHttpRequest && (!((FullHttpRequest) msg).headers().contains(HttpHeaders.UPGRADE))) ```

Reply:
Moved the health check code to top of the method.